services/expo/src/routes/project.ts

```diff
@@ -491,24 +493,35 @@ projectRoutes.route("/special/dashboard").get(
       },
     });
 
+    const hexUrl = req.query?.hexathon ? `/hexathons/${req.query?.hexathon}` : `/hexathons`;
     const hexathons = await apiCall(
       Service.HEXATHONS,
       {
-        url: `/hexathons`,
+        url: hexUrl,
         method: "GET",
+        params: { id: req.query?.hexathon },
       },
       req
     );
 
-    for (const project in projects) {
-      for (const hexathon in hexathons) {
-        if (projects[project].hexathon === hexathons[hexathon].id) {
-          projects[project].hexathon = hexathons[hexathon];
+    if (hexathons.length === undefined) {
```

Comment:
either change the condition to `req.query?.hexathon`, it's more readable that way since we'll know this block pertains to when a single hexathon is returned, or add a brief comment over it